utils/pattern/src/frontend/mod.rs

```diff
@@ -2,27 +2,27 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+extern crate alloc;
```

Comment:
Oops, you shouldn't need this. The bounds on your new impl are wrong. They should be on Store, not Store::Owned.